pkgs/development/libraries/libgit2/default.nix

```diff
@@ -46,14 +47,31 @@ stdenv.mkDerivation (finalAttrs: {
     "-DCMAKE_LIBRARY_PATH=${stdenv.cc.libc}/lib"
   ];
 
-  nativeBuildInputs = [ cmake python3 pkg-config ];
+  nativeBuildInputs = [ cmake python3 pkg-config ] ++ lib.optionals (!stdenv.hostPlatform.isWindows) [
+    # Needed for `git apply`; see `prePatch`
+    buildPackages.gitMinimal
+  ];
 
   buildInputs = [ zlib libssh2 openssl pcre2 llhttp ]
     ++ lib.optional withGssapi krb5
     ++ lib.optional stdenv.hostPlatform.isDarwin Security;
 
   propagatedBuildInputs = lib.optional (!stdenv.hostPlatform.isLinux) libiconv;
 
+  patches = [
+    # Required by nix. Will be included in the next release of libgit2
+    ./patches/libgit2-mempack-thin-packfile.patch
+  ] ++ lib.optionals (!stdenv.hostPlatform.isWindows) [
+    ./patches/libgit2-packbuilder-callback-interruptible.patch
+  ];
+
+  # Only `git apply` can handle git binary patches
+  prePatch = lib.optionalString (!stdenv.hostPlatform.isWindows) ''
+    patch() {
```

Comment:
Should we unset the function in postPatch again?

Reply:
Done.